R/sem.R

```diff
@@ -256,6 +256,20 @@ checkLavaanModel <- function(model, availableVars) {
       break
     }
 
+    if(isFALSE(slot(fit, "optim")$converged)) {
+      errormsg <- gettextf("Estimation failed! Message: Model %s did not converge!", options[["models"]][[i]][["modelName"]])
+      modelContainer$setError(errormsg)
+      modelContainer$dependOn("models")
+      break
+    }
+
+    if(lavaan::fitMeasures(fit, "df") < 0 ) {
+      errormsg <- gettextf("Estimation failed! Message: Model %s has negative degrees of freedom.", options[["models"]][[i]][["modelName"]])
+      modelContainer$setError(errormsg)
+      modelContainer$dependOn("models")
+      break
+    }
```

Comment:
So from the `break` I take it that we crash as soon as one model encounters a major issue. But perhaps it's more user-friendly to still show the results for the models that did work?

@Kucharssim, @JMBKoch any ideas/ suggestions?

Reply:
That’s how it was previously (showing a footnote under the main table that some model did not work), but apparently that was not very clear enough. It would also take a lot of work to rewrite code for all other output such that it does not crash when you wanna use it with invalid model.

I think there are many options to take here, but all have some pros and cons - I did not think of any that would result in a completely clean solution. I think having info/warning boxes would help, but for now I resorted to this

Reply:
> but for now I resorted to this

Yeah, I think this PR is definitely fine as is, but if we'd prefer showing partial output when one model fails then we should make an issue for it (and otherwise we don't).

Reply:
I made an issue here https://github.com/jasp-stats/INTERNAL-jasp/issues/1409